uxarray/io/_exodus.py

```diff
@@ -50,58 +50,54 @@ def _read_exodus(ext_ds, grid_var_names):
         elif key == "coord":
             ds.Mesh2.attrs['topology_dimension'] = INT_DTYPE(
                 ext_ds.dims['num_dim'])
-            ds["Mesh2_node_x"] = xr.DataArray(
-                data=ext_ds.coord[0],
-                dims=["nMesh2_node"],
-                attrs={
-                    "standard_name": "longitude",
-                    "long_name": "longitude of mesh nodes",
-                    "units": "degrees_east",
-                })
-            ds["Mesh2_node_y"] = xr.DataArray(
-                data=ext_ds.coord[1],
-                dims=["nMesh2_node"],
-                attrs={
-                    "standard_name": "latitude",
-                    "long_name": "latitude of mesh nodes",
-                    "units": "degrees_north",
-                })
+            ds["Mesh2_node_x"] = xr.DataArray(data=ext_ds.coord[0],
+                                              dims=["nMesh2_node"],
+                                              attrs={
+                                                  "standard_name": "x",
+                                                  "long_name": "cartesian x",
+                                                  "units": "m",
+                                              })
+            ds["Mesh2_node_y"] = xr.DataArray(data=ext_ds.coord[1],
+                                              dims=["nMesh2_node"],
+                                              attrs={
+                                                  "standard_name": "y",
+                                                  "long_name": "cartesian y",
+                                                  "units": "m",
+                                              })
```

Comment:
Relating to the discussion above, what I'm referring to is that we shouldn't use `Mesh2_node_x/y/z` for storing Cartesian coordinates, these should exlucisvely be using for coordinates that are in degrees.

`Mesh2_node_cart_x/y/z` should be used when reading/writing in Cartesian coordinates. Pigning @hongyuchen1030 if you have any comments on this also, since you wrote `_populate_lonlat_coord()` and similar functions.

Reply:
The above lines are corresponding to the read part, we store the cartesian in `Mesh2_node_x/y/z` and later call `_populate_lonlat_coord()` [after read call coordinate fix](https://github.com/UXARRAY/uxarray/pull/373/files#diff-0278cc295f397962b7270dffc8271f5b39394d97520ac04b5632e1fec2204656R235)

Reply:
I see now, the implementation of `_populate_lonlat_coord()` then seems a bit hacky then, since it's purpose here is essentially "correcting" the improper parsing. We should strongly consider the following approach with some alterations to `_populate_lonlat_coord`: 

At the grid reading step (all grid formats, not just Exodus), we should identify whether the coordinates are in degrees or Cartesiain, and then populate the respective `Mesh2_node_x/y` or `Mesh2_node_cart_x/y/z` instead of assigning invalid coordinates to `Mesh2_node_x/y/z` and then correcting them with `_populate_lonlat_coord`. 

`Mesh2_node_x/y`, which are the default coordinates used throughout UXarray, should always be in degrees, specially with the range `([-180, 180], [-90, 90])` to stay consistent. We should never store Cartesiain coordinates in here, even if they will eventually be "corrected" by `_populate_lonlat_coord`

Reply:
From the beginning of last summer, we already agreed that `Mesh22_node_x/y` should always store the longitude and latitude in degrees. 

But for the range `([-180, 180], [-90, 90])` we haven't had any regulation on our repository yet. My implementation always uses `[0,360)`. I am open to any notation, just keep in mind that the right side of this range should always be opening: we don't want to have two representations for the same degree.

And we want to convert the degree to cartesian as less as possible. We don't want to do casting back and forth repeatly because that might bring impact to the precision as I explained before.